### DISCUSSION TITLE
[SUGGESTION] Player-bot system

### DISCUSSION THREAD

**Discussion by Bogir** [azerothcore/azerothcore-wotlk#3936] [Ideas]
It was a big mistake =)

**Comment by FrancescoBorzi** [comment]
good idea @Bogir  thanks for suggesting it

**Comment by Helias** [comment]
I really like this idea :+1:

**Comment by brussens** [comment]
Hmmm, it's good idea of new module)

**Comment by disclosurez** [comment]
> 
> 
> It's seems bots are works, but there is nobody in "Who"-list
> ![изображение](https://user-images.githubusercontent.com/25766571/68311481-aaf41600-00c2-11ea-935a-acc1c883045d.png)

can you create a guide  how to use these, then if more people use it we can then put them on who list with more devs

**Comment by Bogir** [comment]
How? Just clone, compile and simple run it.

**Comment by BarbzYHOOL** [comment]
Have you tested Bogir?

Other sources:

- https://bitbucket.org/trickerer/trinity-bots  (original author's work I think)
- https://github.com/Psilovybin/azcore-bots attempt at adding the bots

Same topic: https://github.com/azerothcore/forum/issues/33

**Comment by fang139842** [comment]
AZ怎么使用啊

**Comment by Zoidwaffle** [comment]
The project seems well maintained for TC here:

https://github.com/trickerer/Trinity-Bots

**Reply by virgo77** [reply]
it seems! just updated 9 hours ago Oo'
Do you think it can be used as is, in AC? Or it must be 'forked' before? because I would like to test it. But i'm not comfy if it is not 'AC Approved' ^^
Thanks :)

**Comment by trickerer** [comment]
Well, just letting you know, but Trinity-Bots now have [AzerothCore version](https://github.com/trickerer/AzerothCore-wotlk-with-NPCBots). Not a module though (just try to guess how many new hooks it requires).

**Reply by Winfidonarleyan** [reply]
work?

**Reply by virgo77** [reply]
Hey there, thanks for the reply :)

I see that it is an AC 'fork' but not with the same core build number, so I do think we need to download this one which includes the bots and then play. But, is it playable if the DB is not in the same build number as this core? I mean, I update the original AC once a week. So I think that the original build is different from the 'fork' one. And when we update the original core, the DB is updated too. Will the 'fork' be playable instead?

Another question: are these bots capable to play on their own (it would be great! ^^) and that we can meet sometimes on some zones or cities, or is that an 'addon style' to call bots and play with?

Thanks in advance for your answers

++

V.

**Reply by trickerer** [reply]
Hey.

I plan to merge recent changes from upstream once a week or so (just like with TC bots repo I have). So my AC fork will include all the updates the original AzerothCore-wotlk master branch receives.

NPCBots' AI doesn't have any strategic component to it, so - no, they don't travel on their own and don't have any goals other than help players. They are just stationary spawns.
Although if bots of factions hostile to each other are spawned one next to the other, they may engage into, sometimes very long, combat, which may take them to other areas. Still, they will eventually break from it and return to their spawn positions.

**Reply by LukasVolgger** [reply]
Is it possible to use it as a module? Just apply patch and go on?

**Reply by virgo77** [reply]
Hi Trickerer and thanks a lot for your reply :)
So I think I will try it soon, with an export of my actual DB and test it in a fresh install.

I wonder that you maybe need some feedback about NPCs' behavior. So if I test it quickly, I'll try some dungeons I think, or just questing.

Just a question: are the mobs level adaptive, or are they only 80? (maybe 100 to kill Arthas more easily huh huh^^)
Another question (sorry): did you merge once a week, the same day (like Sunday, or Wednesday)?

I stay tuned. Thanks for your job ;)

V.

**Reply by trickerer** [reply]
@LukasVolgger as I said, it's not a module, so it can't be used as one. This mod is way to intrusive, using hooks for that is hard to justify. This is why I have to maintain its compatibility with core.
But yes, IF you use compatible core revision, you can just apply the patch and build.

**Reply by trickerer** [reply]
@virgo77 [here](https://github.com/trickerer/Trinity-Bots/issues/212) is the 'thread' where AC npcbots issues are being gathered. Things got complicated unfortunatelly with new proc system introducing some new issues, delaying the push of npcbots fixes.

Bots' level **is** adaptive. Actually, npcbots mimic players in many aspects, like using LFG or vehicles, not just stats. I recommend going through Readme, it is TLDR-ish but it covers pretty much everything.

Normally, I merge once a week, yes. Saturday 0:00 to 2:00 (UTC+0).
